src/week4/WordBreak.java

```diff
@@ -0,0 +1,22 @@
+package week4;
+
+public class WordBreak {
+
+	public static void main(String[] args) {
+		String[] dictionary = { "pear","salmon","foot","prints"};
+		String input = "salmonloop";
+		System.out.println(wordbreakcheck(dictionary, input));
+	}
+
+	private static boolean wordbreakcheck(String[] dictionary, String input) {
+		if(input.equals("")||(input==null))

```

Comment:
**Hi, AishwaryaPSU**
Thanks for sharing your code! I have some feedback/suggestions for the WordBreak.java solution.  It's broken up depending on how you will handle the null and empty string edge cases.  There was a message from taylor.garren in the week4discussion channel specifying to return false for an empty string, and throw an exception for a null string. I hope you find all the feedback useful, and thanks for looking!
### null: _false_ empty string: _false_
1.  **Null pointer exception: Swap || operands to take advantage of short circuiting**
12: `if ( input == null || input.equals("") ) return false;`
All binary operators except for the assignment operators are evaluated from left to right; A NullPointerException will be thrown on line 12 if the wordbreakcheck method receives a null argument for the 'input' parameter. It crashes when the logical OR operator tries to derefence a null 'input' object. It's essentially trying to run `null.equals("")`. We can fix this by swapping the order of the operands for the logical OR operator. This forces the evaluation of 
`input==null`first; ensuring`input.equals("")`is not evaluated when the input argument is a null object. The `||` operator simply short circuits and skips evaluation of the second operand after the first expression evaluates true. 

2. **Optimization: Use input.isEmpty() instead of input.equals("")**
12: `if ( input==null || input.isEmpty() ) return false;`
`input.equals("")` can be replaced with `input.isEmpty()` for faster runtime performance. Strings are immutable, and store an initialized count variable in their constructor. The isEmpty() method can quickly identify an empty string by accessing and making a comparison between the private array's 'count' variable and zero.  The equals(Object obj) method, in comparison, must first verify the arguments are not the same Object, perform an instanceof evaluation, complete a cast to String, a String length evaluation, and finally iterate over the string if they were equal in length. 
_**Consideration:** This might break things if using java 1.5 and below._

### null:_Exception_ empty string: _false_
1. **Add 'throws InvalidArgumentException' to method signature**
12: `private static boolean wordbreakcheck(String[] dictionary, String input) throws InvalidArgumentException {`
13: 'if ( input == null || dictionary == null ) throw new InvalidArgumentException();
14: 'if ( input.isEmpty() || dictionary.length == 0 ) return false;
Additional logic is necessary to handle null and empty strings differently.  

> https://docs.oracle.com/javase/tutorial/java/nutsandbolts/operators.html
> https://stackoverflow.com